src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -343,70 +613,66 @@ define(function (require, exports, module) {
             }
 
             return (
-                <li className={classnames(layerClasses)}>
-                    <div
-                        style={dragStyle}
-                        className={classnames(faceClasses)}
-                        data-layer-id={layer.id}
-                        data-kind={layer.kind}
-                        onMouseDown={!this.props.disabled && this.props.handleDragStart}
-                        onClick={!this.props.disabled && this._handleLayerClick}>
-                        <Button
-                            title={tooltipTitle + tooltipPadding}
-                            disabled={this.props.disabled}
-                            className={classnames("face__kind", iconClassModifier)}
-                            data-kind={layer.isArtboard ? "artboard" : layer.kind}
-                            onClick={this._handleIconClick}
-                            onDoubleClick={this._handleLayerEdit}>
-                            <SVGIcon
-                                CSSID={iconID}
-                                viewbox="0 0 24 24"/>
-                        </Button>
-                        <span className="face__separator">
-                            <TextInput
-                                title={layer.name + tooltipPadding}
-                                className="face__name"
-                                ref="layerName"
-                                type="text"
-                                value={layer.name}
-                                editable={!this.props.disabled && nameEditable}
-                                preventHorizontalScrolling={true}
-                                onKeyDown={this._skipToNextLayerName}
-                                onChange={this._handleLayerNameChange}>
-                            </TextInput>
-                            {showHideButton}
-                        </span>
-                        <ToggleButton
-                            disabled={this.props.disabled}
-                            title={strings.TOOLTIPS.LOCK_LAYER + tooltipPadding}
-                            className="face__button_locked"
-                            size="column-2"
-                            buttonType={layer.locked ? "toggle-lock" : "toggle-unlock"}
-                            selected={layer.locked}
-                            onClick={this._handleLockToggle}>
-                        </ToggleButton>
-                    </div>
-                </li>
+                <Draggable
+                    type="layer"
+                    target={this.props.layer}
+                    beforeDragStart={this._handleBeforeDragStart}
+                    onDragStart={this._handleDragStart}
+                    onDrag={this._handleDrag}
+                    onDragStop={this._handleDragStop}>
+                <Droppable
```

Comment:
Sorry for only commenting on whitespace, but did you intend for this to violate the usual indentation rules? I can imagine why you might do so intentionally...